### PR TITLE
Update auth policies to use new AuthContext format

### PR DESCRIPTION
**Pull request by IsuruGunarathne**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

 

## Summary by CodeRabbit

* **Refactor**
  * Consolidated authentication state into a unified AuthContext (including AuthType and PolicyName) across auth and authz policies; success/failure now populate this shared context instead of legacy metadata.

* **Tests**
  * Expanded and updated tests for Basic Auth, JWT, MCP, MCP AuthZ and API Key policies to validate AuthContext-driven behavior, claim/scope handling, and many edge cases.

* **Chores**
  * Bumped policy-definition versions and updated module dependencies across affected policies.

* **Behavior**
  * API key handling updated to use hashed key storage/validation.

### CODE REVIEW

File: policies/api-key-auth/apikey.go [hunk 1]
```diff
@@ -184,14 +179,11 @@ func (p *APIKeyPolicy) handleAuthSuccess(ctx *policy.RequestContext) policy.Requ
...
-	ctx.Metadata[MetadataKeyAuthSuccess] = true
-	ctx.Metadata[MetadataKeyAuthMethod] = "api-key"
-
-	slog.Debug("API Key Auth Policy: Authentication metadata set",
-		"authSuccess", true,
-		"authMethod", "api-key",
-	)
+	ctx.SharedContext.AuthContext = &policy.AuthContext{
+		Authenticated: true,
+		AuthType:      "apikey",
```
Comment (renuka-fernando) [inline policies/api-key-auth/apikey.go]: shall we use consts for this

File: policies/api-key-auth/apikey.go [hunk 2]
```diff
@@ -184,14 +179,11 @@ func (p *APIKeyPolicy) handleAuthSuccess(ctx *policy.RequestContext) policy.Requ
...
+		PolicyName:    "api-key-auth",
```
Comment (renuka-fernando) [inline policies/api-key-auth/apikey.go]: This too

File: policies/api-key-auth/apikey.go (hunk 1 above)
Comment (IsuruGunarathne) [reply policies/api-key-auth/apikey.go]: Updated in https://github.com/wso2/gateway-controllers/pull/62/commits/3e30a6244f754348209f16aba136480a175be456

File: policies/api-key-auth/apikey.go (hunk 2 above)
Comment (IsuruGunarathne) [reply policies/api-key-auth/apikey.go]: Updated in https://github.com/wso2/gateway-controllers/pull/62/commits/3e30a6244f754348209f16aba136480a175be456

File: policies/jwt-auth/jwtauth.go [hunk 3]
```diff
@@ -1276,46 +1266,34 @@ func parseScopes(scopeClaim, scpClaim interface{}) []string {
...
+		Issuer:        iss,
+		Audience:      parseAudience(claims["aud"]),
+		Scopes:        buildScopesMap(claims),
+		Properties:    buildProperties(claims),
 	}
 
+	slog.Debug("JWT Auth Policy: AuthContext set",
+		"subject", subject,
+		"issuer", iss,
+	)
```
[the remark this answers is not in the document]
Comment (IsuruGunarathne) [reply policies/jwt-auth/jwtauth.go:1297]: This is inline with the existing logging behaviour

File: policies/api-key-auth/apikey.go [hunk 4]
```diff
@@ -184,14 +184,11 @@ func (p *APIKeyPolicy) handleAuthSuccess(ctx *policy.RequestContext) policy.Requ
...
 
-	// Set metadata indicating successful authentication
-	ctx.Metadata[MetadataKeyAuthSuccess] = true
-	ctx.Metadata[MetadataKeyAuthMethod] = "api-key"
-
-	slog.Debug("API Key Auth Policy: Authentication metadata set",
-		"authSuccess", true,
-		"authMethod", "api-key",
-	)
+	ctx.SharedContext.AuthContext = &policy.AuthContext{
```
Comment (renuka-fernando) [inline policies/api-key-auth/apikey.go:186]: We have to check if an already values is there in the ctx.SharedContext.AuthContext and if so move it to AuthContext.Previous right?

Comment (IsuruGunarathne) [reply policies/api-key-auth/apikey.go:186]: Yes, implemented in https://github.com/wso2/gateway-controllers/pull/62/commits/5420ae697d96a2d59384728a0d581df02d443916

**Review by renuka-fernando — APPROVED** [review]
